Sources/TestingMacros/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+cmake_minimum_required(VERSION 3.28)
+
+if(POLICY CMP0157)
+  cmake_policy(SET CMP0157 NEW)
+endif()
+
+project(TestingMacros
+  LANGUAGES Swift)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+include(FetchContent)
+set(FETCHCONTENT_BASE_DIR ${CMAKE_BINARY_DIR}/_d)
+FetchContent_Declare(SwiftSyntax
+  GIT_REPOSITORY https://github.com/apple/swift-syntax
+  GIT_TAG main)
+FetchContent_MakeAvailable(SwiftSyntax)
+
+include(Settings)
+add_executable(TestingMacros
+  ConditionMacro.swift
+  SuiteDeclarationMacro.swift
+  Support/Additions/DeclGroupSyntaxAdditions.swift
+  Support/Additions/EditorPlaceholderExprSyntaxAdditions.swift
+  Support/Additions/FunctionDeclSyntaxAdditions.swift
+  Support/Additions/MacroExpansionContextAdditions.swift
+  Support/Additions/TokenSyntaxAdditions.swift
+  Support/Additions/TriviaPieceAdditions.swift
+  Support/Additions/TypeSyntaxProtocolAdditions.swift
+  Support/Additions/VersionTupleSyntaxAdditions.swift
+  Support/Additions/WithAttributesSyntaxAdditions.swift
+  Support/Argument.swift
+  Support/AttributeDiscovery.swift
+  Support/AvailabilityGuards.swift
+  Support/CommentParsing.swift
+  Support/ConditionArgumentParsing.swift
+  Support/CRC32.swift
+  Support/DiagnosticMessage.swift
+  Support/DiagnosticMessage+Diagnosing.swift
+  Support/SourceCodeCapturing.swift
+  Support/SourceLocationGeneration.swift
+  TagMacro.swift
+  TestDeclarationMacro.swift
+  TestingMacrosMain.swift)
+set_target_properties(TestingMacros PROPERTIES
+  ENABLE_EXPORTS TRUE)
```

Comment:
Why does this executable target need to enable exports?

Reply:
Without this, I was seeing symbols exported from TestingMacros.exe like this:
```
$s4main11ExpectMacroVN
type metadata for main.ExpectMacro
```

I believe this property sets `-rdynamic` which gets us the correct export:
```
$s13TestingMacros11ExpectMacroVN
type metadata for TestingMacros.ExpectMacro
```

Reply:
Hmm, yeah, this was a bug in CMake 3.29 and 3.29.1. It was fixed in 3.29.2. We could say that we don't support 3.29.0 and 3.29.1 and not have this, which would be better.

The property does set `-rdynamic` and generates a swiftmodule for the macro, exporting all of symbols in the executable.